include/seqan3/io/detail/record.hpp

```diff
@@ -125,4 +135,21 @@ struct select_types_with_ids<field_types, field_types_as_ids, selected_field_ids
 };
 //!\endcond
 
+
+// ----------------------------------------------------------------------------
+// get_or_ignore
+// ----------------------------------------------------------------------------
+
+/*!\brief Perform a get-by-field on the record and return reference to std::ignore if record doesn't have field.
+ * \ingroup core
+ */
+template <field f, typename field_types, typename field_ids>
+auto & get_or_ignore(record<field_types, field_ids> & r)
```

Comment:
no, because we want to access the elements of the record so that we can write to them. You can write to const elements, and you don't care about the elements if they go out of scope after the function returns.